### PR TITLE
Feature/us7426 refactor unit tests

### PR DESCRIPTION
**Pull request by ilpadre**

### CODE REVIEW

File: src/app/components/add-me-to-map/add-me-to-map.component.spec.ts [hunk 1]
```diff
@@ -68,7 +97,20 @@ describe('Component: Add Me to the Map', () => {
         UserLocationService,
         AddressService,
         ToastsManager,
-        ToastOptions
+        ToastOptions,
```
Comment (JoeKer1) [inline src/app/components/add-me-to-map/add-me-to-map.component.spec.ts]: ToastsManager should be mocked, then you shouldn't need ToastOptions. 
A lot of these services provided shouldn't be necessary.

File: src/app/components/add-me-to-map/add-me-to-map.component.spec.ts [hunk 2]
```diff
@@ -68,7 +97,20 @@ describe('Component: Add Me to the Map', () => {
         UserLocationService,
         AddressService,
```
Comment (JoeKer1) [inline src/app/components/add-me-to-map/add-me-to-map.component.spec.ts]: Address service should be mocked.

File: src/app/components/add-me-to-map/add-me-to-map.component.spec.ts [hunk 3]
```diff
@@ -27,14 +26,40 @@ import { ContentService } from 'crds-ng2-content-block/src/content-block/content
...
+    mockUserLocationServicee,
+    mockLocationService;
+
 
   beforeEach(() => {
     mockContentService = jasmine.createSpyObj<ContentService>('content', ['loadData', 'getContent']);
     mockLocation = jasmine.createSpyObj<Location>('location', ['back']);
+    mockContentService = jasmine.createSpyObj<ContentService>('content', ['loadData']);
+
+    mockAddMeToTheMapHelperService = jasmine.createSpyObj<AddMeToTheMapHelperService>('AddMeToTheMapHelperService', ['constructor']);
```
Comment (JoeKer1) [inline src/app/components/add-me-to-map/add-me-to-map.component.spec.ts]: I don't think hlpr is used in the component at all (unless i'm behind) so probably should be removed from the constructor and not mocked.

File: src/app/components/add-me-to-map/add-me-to-map.component.spec.ts [hunk 4]
```diff
@@ -27,14 +26,40 @@ import { ContentService } from 'crds-ng2-content-block/src/content-block/content
...
 import { Location } from '@angular/common';
+import { LocationService } from '../../services/location.service';
 
 describe('Component: Add Me to the Map', () => {
 
-  let mockContentService, mockLocation;
+  let mockLocation;
+  let mockAddMeToTheMapHelperService,
+    mockBlandPageService,
+    mockCookieService,
```
Comment (JoeKer1) [inline src/app/components/add-me-to-map/add-me-to-map.component.spec.ts]: I don't think cookieService needs to be mocked, not 100% sure.

File: src/app/components/authentication/authentication.component.spec.ts [hunk 5]
```diff
@@ -24,27 +20,35 @@ describe('Component: Authentication', () => {
...
-        CookieService,
-        SessionService,
-        StateService,
         { provide: LoginRedirectService, useValue: mockLoginRedirectService },
-        PinService,
         FormBuilder,
-        StoreService,
-        RouterTestingModule,
-        BlandPageService
+        SessionService,
```
Comment (JoeKer1) [inline src/app/components/authentication/authentication.component.spec.ts]: I think session Service should be mocked here, also i don't think LoginRedirectService should be injected like it is being done here. That is a pattern for the actual service test.

File: src/app/components/add-me-to-map/add-me-to-map.component.spec.ts [hunk 6]
```diff
@@ -12,34 +11,46 @@ import { AgmCoreModule } from 'angular2-google-maps/core';
...
+    mockSessionService = jasmine.createSpyObj<SessionService>('sessionService', ['constructor']);
+    mockStateService = jasmine.createSpyObj<StateService>('stateService', ['constructor']);
+    mockContentService = jasmine.createSpyObj<ContentService>('contentService', ['loadData']);
+    mockAddressService = jasmine.createSpyObj<AddressService>('addressService', ['constructor']);
+    mockToastsManager = jasmine.createSpyObj<ToastsManager>('toastsManager', ['constructor']);
+
 
     TestBed.configureTestingModule({
       declarations: [
         AddMeToMapComponent, AddressFormComponent,
```
Comment (JoeKer1) [inline src/app/components/add-me-to-map/add-me-to-map.component.spec.ts]: AddressFormComponent should be MockComponent({ selector: 'address-form', inputs: ['whateverTheyAre'] }); You may be able to get rid of additional providers after doing this.

File: src/app/components/authentication/authentication.component.spec.ts [hunk 7]
```diff
@@ -69,24 +68,24 @@ describe('Component: Authentication', () => {
   }));
 
   it('should set the "navigatedBackFromAuthComponent" prop on state service',
-      inject([StateService], (stateService) => {
-    stateService.navigatedBackFromAuthComponent = false;
-    comp.back();
-    expect(stateService.navigatedBackFromAuthComponent).toEqual(true);
-  }));
+    inject([StateService], (stateService) => {
```
Comment (JoeKer1) [inline src/app/components/authentication/authentication.component.spec.ts]: I still don't think services should be injected like this for component tests. Maybe it's just the nature of this test?

Comment (ilpadre) [inline src/app/components/authentication/authentication.component.spec.ts]: I think you're right. I just didn't see this one.

File: src/app/components/add-me-to-map/add-me-to-map.component.spec.ts [hunk 8]
```diff
@@ -82,7 +83,7 @@ describe('Component: Add Me to the Map', () => {
 
   it('should attach class selector to body element after view init', () => {
     // Start fresh...
-    document.querySelector('body').classList.remove('modal-open');
+    document.querySelector('body').classList.remove('modal-open');//
```
Comment (Doug-Shannon) [inline src/app/components/add-me-to-map/add-me-to-map.component.spec.ts]: extra slashes

**Pull request merged**